contracts/utils/Swap.sol

```diff
@@ -329,13 +320,16 @@ contract Swap is ISwap, ReentrancyGuard, Ownable {
      * @param swapId unique identifier of the swap offer
      */
     function averagePriceForOffer(uint256 swapId)
-        public
+        external
         view
         override
         returns (uint256)
     {
         Offer storage offer = swapOffers[swapId];
-        require(offer.seller != address(0), "Offer does not exist");
+        require(
+            offer.totalSize != 0,
+            "Offer has not settled or offer does not exist"
```

Comment:
The wording here is off, should be "Offer has settled or offer does not exist" instead

Reply:
Changed this to "Offer does not exist". I just checked that this value is static after offer creation and we don't set it to 0 after settlement because we need it in the calculation.